feat(ui-kit): SideMenu component UI , test and storybook are added.

manishRise11 opened the pull request:
### Student variant
 
[On figma](https://www.figma.com/design/8KEwRuOoD5IgxTtFAtLlyS/Just_Do_Ad-1.2?node-id=5272-92486&t=IrCukPhd3jcco8aU-4)

- expanded:

![Image](https://github.com/user-attachments/assets/6c53b6eb-27f5-43aa-bad1-a76770754de2)

- collapsed: note it grows vertically with the page

![Image](https://github.com/user-attachments/assets/dd5711ba-74ba-40bb-8c35-e86f05d5f8df)

### Coach variant / Course creator variant (has the additional badge next to "Coach"):

[On figma](https://www.figma.com/design/8KEwRuOoD5IgxTtFAtLlyS/Just_Do_Ad-1.2?node-id=5272-92096&t=IrCukPhd3jcco8aU-4)

- expanded:

![Image](https://github.com/user-attachments/assets/b4e45258-39df-4bd5-96e9-b4b04b050a35)

- collapsed:

![Image](https://github.com/user-attachments/assets/d45c6507-0604-4638-9c66-802881a6860d)
---

## Checklist

- [x] I have run the lint, build, and test commands locally, as explained in the README, before creating this PR to ensure the code is clean and functional.
- [x] My changes align with the **core models** that have been discussed and agreed upon.
- [x] The component design has been compared to the **Figma design** to ensure consistency.
- [x] The **component is fully responsive** and behaves as expected across different screen sizes.
- [x] The component **supports keyboard web accessibility** (if applicable).
- [x] I have written **unit tests** or **integration tests** to cover the component's functionality above the coverage threshold.
- [x] I have updated the **documentation** for the component, including usage examples and guidelines.
- [x] I have **commented** my code, particularly in complex or non-obvious areas.
- [x] The **component is properly exported** for use in other parts of the application.

---

## Related Issues

<!-- Link any related issue or task. For example: -->
Fixes #73

alebg commented on packages/ui-kit/lib/components/icons/icon-log-out.tsx:
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
```
@manishRise11 why is React imported here?

alebg commented on packages/ui-kit/lib/components/sidemenu/sidemenu.tsx:
```diff
@@ -0,0 +1,185 @@
+'use client';
```
@manishRise11 why is this marked as a client component? as far as I can see, it looks like its a perfectly OK server side component. Correct me if I'm wrong though

alebg commented on packages/ui-kit/lib/components/sidemenu/sidemenu.tsx:
```diff
@@ -0,0 +1,185 @@
+        ))}
+      </div>
+
+      {/* Collapse Button */}
```
@manishRise11 when the component is collapsed, I can't click on the last icon. In your story, it's the logout button. The 'expand' icon and function take the whole space and there's no way to click the logout button

![immagine](https://github.com/user-attachments/assets/73cf9135-ecf4-4fae-bb0f-f484f568d217)

@carlapbg I guess the easiest would be to just add more space between the collapse/expand icon and the last menu item, right?

carlapbg replied:
@manishRise11 I still have that problem. Make sure only the button itself can be clicked and not the entire area around it.

alebg commented on packages/ui-kit/lib/components/sidemenu/sidemenu.tsx:
```diff
@@ -0,0 +1,185 @@
+      onClick={isCollapsed ? () => onClickToggle(isCollapsed) : undefined}
+      data-testid="menu-container"
+    >
+      {/* User Profile Section */}
```
@manishRise11 TBD but this could also be a layout component, so the app can decide what to pass exactly. Don't change anything yet, I'll come back to you

manishRise11 commented:
Hi @alebg Sir,

I have implemented the following changes:

- Removed unnecessary React import from `icon-log-out.tsx`.

- Removed 'use directive from `sidemenu.tsx` since the toggle logic has been refactored into a callback, making the component suitable for server usage.

- In `sidemenu.tsx`, updated styling for the collapsed state:
  - Applied `h-full` to match Figma designs and ensure consistent UI.

- Improved UX for collapsed state:
  - Made the entire collapsed sidebar menu act as the expand toggle button.
  - This removes the need to scroll to the bottom to find the toggle icon (previously positioned at bottom-right).
  - Trade-off: Individual menu items/icons cannot be selected or clicked when collapsed.

- Added spacing/gap between menu items and the toggle icon to improve layout clarity.

- Added an explicit `toBeInTheDocument()` assertion in `sidemenu.test.tsx` to improve test robustness.

- Updated the `userRole` in the Coach story (Storybook) from `'courseCreator'` to `'coach'` to match the badge display design.

---

Questions:

1. Is it acceptable that the collapsed side menu acts as a toggle (disabling individual item interactions)?
2. Or should we keep the individual icons clickable and redesign the toggle interaction or placement to avoid scroll-related issues?

Additionally, the expanded side menu uses `h-auto` while the collapsed one uses `h-full`. Please advise if this layout handling is appropriate or needs adjustments.

carlapbg reviewed (changes requested):
Hi! Just some things about this component:

- Let’s use the [nav-link states](https://www.figma.com/design/8KEwRuOoD5IgxTtFAtLlyS/Just_Do_Ad-1.2?node-id=5255-54805&t=XIoUqMrtIwTGZrf9-4) in this sideMenu. (Default, hover and active):
![Screenshot 2025-04-19 at 15 26 43](https://github.com/user-attachments/assets/6ae4a1f4-794a-4dfe-a58a-0b686402117f)

- When the menu is collapsed, we should still be able to click in the icons. You can add a hover on each icon indicating each title (dashboard, courses, calendar…) 

- Right now if the name is too long the menu just grows. The width should be fixed. And if the name is too long go can also add a hover.

manishRise11 commented:
> Hi! Just some things about this component:
> 
> * Let’s use the [nav-link states](https://www.figma.com/design/8KEwRuOoD5IgxTtFAtLlyS/Just_Do_Ad-1.2?node-id=5255-54805&t=XIoUqMrtIwTGZrf9-4) in this sideMenu. (Default, hover and active):
>   ![Screenshot 2025-04-19 at 15 26 43](https://private-user-images.githubusercontent.com/74186806/435396307-6ae4a1f4-794a-4dfe-a58a-0b686402117f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDUwNzEwMjQsIm5iZiI6MTc0NTA3MDcyNCwicGF0aCI6Ii83NDE4NjgwNi80MzUzOTYzMDctNmFlNGExZjQtNzk0YS00ZGZlLWE1OGEtMGI2ODY0MDIxMTdmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE5VDEzNTIwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNiZjE2NGY1OTg3ZmU3NjcxYjhhZTQ2YzIxYzZlZmEwNGFiZDdkMTZmYWIxOTI1OGI1NmI3ZmVlNmY4YTFkZmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.c5vwWm2XgMiUNtxVnwGFv6hwY62majuzwOklwNvbvN0)
> * When the menu is collapsed, we should still be able to click in the icons. You can add a hover on each icon indicating each title (dashboard, courses, calendar…)
> * Right now if the name is too long the menu just grows. The width should be fixed. And if the name is too long go can also add a hover.

@carlapbg Mam, when I click on the side menu items in Figma, it shows the 'vertical-Xnav-link' component is being used for those items. However, I couldn't find 'vertical-Xnav-link' as a separate component.. Should I use the 'vertical-Xnav-link' variant for implementing the side menu states and interactions you mentioned, or is there a different recommended approach? Please clarify which component or variant you prefer for the side menu implementation.

alebg commented on packages/ui-kit/lib/components/sidemenu/sidemenu.tsx:
```diff
@@ -0,0 +1,184 @@
+        )}
+      </div>
+
+      {/* Menu Groups */}
```
@manishRise11 one question I have: so just by looking at the code, I can't figure out where are the line separators coming from?


![immagine](https://github.com/user-attachments/assets/fa297cd0-66e5-47df-8f3d-cc6e91d2e718)

manishRise11 replied:
@alebg, sir, I am using this border top for each group of items.
<img width="1067" alt="Screenshot 2025-04-20 at 8 07 15 AM" src="https://github.com/user-attachments/assets/2332746a-8017-4450-97b4-e0e3ff35f8d8" />

carlapbg commented:
> > Hi! Just some things about this component:
> > 
> > * Let’s use the [nav-link states](https://www.figma.com/design/8KEwRuOoD5IgxTtFAtLlyS/Just_Do_Ad-1.2?node-id=5255-54805&t=XIoUqMrtIwTGZrf9-4) in this sideMenu. (Default, hover and active):
> >   ![Screenshot 2025-04-19 at 15 26 43](https://private-user-images.githubusercontent.com/74186806/435396307-6ae4a1f4-794a-4dfe-a58a-0b686402117f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDUwNzEwMjQsIm5iZiI6MTc0NTA3MDcyNCwicGF0aCI6Ii83NDE4NjgwNi80MzUzOTYzMDctNmFlNGExZjQtNzk0YS00ZGZlLWE1OGEtMGI2ODY0MDIxMTdmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE5VDEzNTIwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNiZjE2NGY1OTg3ZmU3NjcxYjhhZTQ2YzIxYzZlZmEwNGFiZDdkMTZmYWIxOTI1OGI1NmI3ZmVlNmY4YTFkZmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.c5vwWm2XgMiUNtxVnwGFv6hwY62majuzwOklwNvbvN0)
> > * When the menu is collapsed, we should still be able to click in the icons. You can add a hover on each icon indicating each title (dashboard, courses, calendar…)
> > * Right now if the name is too long the menu just grows. The width should be fixed. And if the name is too long go can also add a hover.
> 
> @carlapbg Mam, when I click on the side menu items in Figma, it shows the 'vertical-Xnav-link' component is being used for those items. However, I couldn't find 'vertical-Xnav-link' as a separate component.. Should I use the 'vertical-Xnav-link' variant for implementing the side menu states and interactions you mentioned, or is there a different recommended approach? Please clarify which component or variant you prefer for the side menu implementation.

Hi! Please use the 'vertical-Xnav-link' variant and apply the design for the different nav-link states

manishRise11 commented:
@carlapbg  mam , @alebg sir , I have incorporated all these changes :-

- I used the vertical-Xnav-link variant (default/hover/active states).
- Collapsed menu: Clickable icons + hover tooltips.
- Fixed width; truncated labels show full text on hover.

- Restructured menu items as children.


Matched Figma design.
Added hover states in both collapsed/expanded modes.

alebg reviewed (approved):
@manishRise11 good job! Thanks a lot. Note I've send a small commit with a bit of formatting, and removing the toggle alert in the story. Let's wait for @carlapbg to give a look, also about the decision we took of not expanding the side menu vertically

carlapbg reviewed (dismissed):
Looks good! Also making the side menu sticky to the left, so it's always there without growing vertically. Thanks!

manishRise11 commented:
> Looks good! Also making the side menu sticky to the left, so it's always there without growing vertically. Thanks!

@alebg  sir and @carlapbg ma'am, I’ve made the changes — replaced h-full with h-fit when the side menu is collapsed, so it maintains the same height as in the expanded state.
